storage/scratchpad/src/sparse_merkle/mod.rs

```diff
@@ -189,12 +189,12 @@ where
     }
 
     /// Constructs a new Sparse Merkle Tree as if we are updating the existing tree multiple times
-    /// with `update_batch`. The function will return the root hash of each individual update and
+    /// with `serial_update`. The function will return the root hash of each individual update and
     /// a Sparse Merkle Tree of the final state.
     ///
-    /// The `update_batch` will take in a reference of value instead of an owned instance. This is
+    /// The `serial_update` will take in a reference of value instead of an owned instance. This is
     /// because it would be nicer for future parallelism.
-    pub fn batch_update(
+    pub fn serial_update(
```

Comment:
We can speed this up by calling `batch_update()` inside, to get a little gain out of your implementation today, but more importantly validate the new implementation in prod as soon as possible.